doc/DatabaseOperations.md

```diff
@@ -0,0 +1,154 @@
+
+## APIs
+
+The library consists of a single fluid component that establishes a
+connection to the data base (host, port, etc.) and provides an interface to
+make PostgresSQL queries
+
+### `fluid.postgresdb.request`
+
+The request component is for initializing the connection with the database and
+provides a means to make queries of it.
+
+#### Component options
+
+These options are all set to `null` in the base grade.  Integrators need to set
+these as appropriate to their situation.
+
+| Option            | Type       | Description | Default |
+| ----------------- | ---------- | ----------- | ------- |
+| `databaseName`    | String     | Required. Name of the database containing the tables, e.g., `"fluid_prefsdb"` | `null` |
+| `host`            | String     | Required. The host for the requests, e.g., `"localhost"` | `null` |
+| `port`            | Integer    | Required. The port for the requests, e.g., `5432` | `null` |
+| `user`            | String     | Required. User administrator's name, e.g., `"admin"`| `null` |
+| `password`        | String     | Required for secure implementations. User administrator's password | `null` |
+
+#### Component members
+
+| Member            | Type       | Description |
+| ----------------- | ---------- | ----------- |
+| `pool`            | Object     | Access to the node-postgres instance.
+
+When initialized, the request component allocates a node-postgress [`Pool`](https://node-postgres.com/api/pool)
+and uses it to establish the connection to the database as per the request's
+options.  It is then ready for database queries.
+
+#### Operations API
+
+##### `query(queryString)` (Invoker)
+
+- `queryString {String}` SQL query string.
+- Returns: `{Promise}` whose value is the result(s) of the query.
+
+The `queryString` can be a single SQL query, or a set of them.  An example of
+the former is:
+
+``` .sql
+SELECT * FROM users WHERE "userId" = 'carla';
+```
+
+An example of the latter is:
+
+``` .sql
+DO $$
+    BEGIN
+        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname='prefsSafesType') THEN
+            CREATE TYPE "prefsSafesType" AS ENUM ('snapset', 'user');
+        END IF;
+        IF NOT EXISTS (SELECT 1 FROM pg_type WHERE typname='accessType') THEN
+            CREATE TYPE "accessType" AS ENUM ('public', 'private', 'shared by trusted parties');
+        END IF;
+    END
+$$;
+CREATE TABLE "prefsSafes" (
+    "prefsSafesId" VARCHAR(36) PRIMARY KEY NOT NULL,
+    "safeType" "prefsSafesType" NOT NULL,
+    name VARCHAR(64),
+    password VARCHAR(64),
+    email VARCHAR(32),
+    preferences JSONB NOT NULL DEFAULT '{}'
+);
+INSERT INTO "prefsSafes" ("prefsSafesId", "safeType", name, password, email)
+    VALUES ('prefsSafe-1', user, 'carla', 'null', 'someone@somewhere.com')
+    RETURNING *
+;
+
+```
+
+In both cases, the query is a single `String`.
+
+The returned `Promise` is configured with a rejection handler that logs any
+error.
+
+##### `bulkQuery(queryArray)` (Invoker)
```

Comment:
Probably rename the function to `executeSQLsFromArray()`?

Reply:
Okay, how about `runSqlArray()`?

Reply:
`runSqlArray()` sounds good.